syntax = "proto3";

package syft.core.tensor;

import "proto/core/tensor/tensor.proto";
import "proto/lib/numpy/array.proto";
import "proto/core/tensor/smpc/party.proto";

message ShareTensor {
  oneof data {
    Tensor tensor = 1;
    syft.lib.numpy.NumpyProto array = 2;
  };

  uint32 rank = 3;
  uint32 nr_parties = 4;
  int32 seed_przs = 5;
  repeated Party parties_info = 6;
}
